### DISCUSSION TITLE
Access to trade information

### DISCUSSION THREAD

**Discussion by anderMartin92** [kernc/backtesting.py#735] [Q&A] [answered]
Hello,

I am trying to collect the data of each trade within the backtesting results.

Reading the documentation, I have seen that the trades have a "pl_pct" attribute that reflects the profit or loss in percentage of each trade. I'm right?

However, when I try to collect the trade data treating it as a dataframe: stats[_trade] I don't find the column referring to that property. How can I find that type of data (pl, pl_pct...) of each trade once the backtesting is done?
Thank you

**Comment by kernc** [comment, marked as answer]
[`Trade.pl_pct`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Trade.pl_pct) is matched by `ReturnPct` in [trade data dataframe](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data):
https://github.com/kernc/backtesting.py/blob/65f54f6819cac5f36fd94ebf0377644c62b4ee3d/backtesting/_stats.py#L65

**Reply by anderMartin92** [reply]
Good! Thanks for the answer, I have realized that it is not the data that I really need.

This data returns the profit percentage of the operation in relation to the value of the position. However, I would like to get the percentage of profit relative to the current balance at the time of closing the trade.

**Reply by kernc** [reply]
That's a simple matter of:
```py
trades = stats['_trades']
profit_pct = trades['PnL'] / stats['_equity_curve']['Equity'].iloc[trades['EntryBar']]`

**Reply by anderMartin92** [reply]
thank you bro!
